Eatery Blue/UI/Favoriting/FavoritesItemsView.swift

```diff
@@ -106,36 +111,49 @@ class FavoritesItemsView: UIView {
+    private func makeDataSource() -> DataSource {
+        let dataSource = DataSource(tableView: tableView) { [weak self] (tableview, indexPath, row) in
+            guard let self else { return UITableViewCell() }
+
+            switch row {
+            case .item(let item, let expanded):
+                guard let cell = tableview.dequeueReusableCell(withIdentifier: FavoritesItemsTableViewCell.reuse, for: indexPath) as? FavoritesItemsTableViewCell else { return UITableViewCell() }
+                cell.configure(item: item, expanded: expanded, itemData: itemData[item.itemName ?? ""])
+                cell.selectionStyle = .none
+                return cell
+            case .label(let text):
+                guard let cell = tableview.dequeueReusableCell(withIdentifier: ClearTableViewCell.reuse, for: indexPath) as? ClearTableViewCell else { return UITableViewCell() }
+                let label = UILabel()
+                label.text = text
+                label.font = .preferredFont(for: .title2, weight: .semibold)
+                let container = ContainerView(content: label)
+                container.layoutMargins = UIEdgeInsets(top: 16, left: 16, bottom: 8, right: 16)
+                cell.configure(content: container)
+                cell.selectionStyle = .none
+                return cell
+            }
+        }
 
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return sortedFavorites.isEmpty ? 1 : sortedFavorites.count
+        return dataSource
     }
-    
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-
-        if sortedFavorites.isEmpty {
-            let label = UILabel()
-            label.text = "No items found..."
-            label.font = .preferredFont(for: .title2, weight: .semibold)
-            let container = ContainerView(content: label)
-            container.layoutMargins = UIEdgeInsets(top: 16, left: 16, bottom: 8, right: 16)
-
-            let cell = ClearTableViewCell(content: container)
-            cell.selectionStyle = .none
-            return cell
-        }
 
-        guard let cell = tableView.dequeueReusableCell(withIdentifier: FavoritesItemsTableViewCell.reuse, for: indexPath) as? FavoritesItemsTableViewCell else { return UITableViewCell() }
-        cell.selectionStyle = .none
-        cell.configure(item: sortedFavorites[indexPath.row], expanded: expanded.contains(indexPath.row), itemData: itemData[sortedFavorites[indexPath.row].itemName ?? ""])
+    /// Updates the table view data source, and animates if desired
+    private func applySnapshot(animated: Bool = true, animation: UITableView.RowAnimation = .fade) {
+        var snapshot = Snapshot()
+        snapshot.appendSections([.main])
+        snapshot.appendItems(
+            (sortedFavorites.isEmpty ? [.label("No items found")] : sortedFavorites.map({ .item($0, expanded.contains(sortedFavorites.firstIndex(of: $0) ?? 0)) }))
```

Comment:
Nitpick: With the map, I would remove the parenthese surrounding the brackets they are not needed, you can directly just do `map {...}` if you are planning to use $0 notation